### ISSUE TITLE
In multiclass text classification, Running loss changes upon changing label values order.

### ISSUE THREAD

**Issue by KrishnaPriyaIITR** [ThilinaRajapakse/simpletransformers#347]
def labelconv(x):
  if x=='Depression':return 0
  elif x=='Alcohol':return 1
  elif x=='Suicide':return 2
  elif x=='Drugs':return 3
Gives:
Running loss: 1.163088
Running loss: 0.793943

whereas if i assign
def labelconv(x):
  if x=='Depression':return 3
  elif x=='Alcohol':return 2
  elif x=='Suicide':return 1
  elif x=='Drugs':return 0
Gives:
Running loss: 1.137920
Running loss: 1.068706

**Comment by ThilinaRajapakse** [comment]
Running loss is the loss at a given training step. Since training is done through random sampling of the training data, it is highly unlikely that you will get the same running losses in two runs. Indeed, something is likely going wrong _if_ you do get the same running losses.

**Comment by KrishnaPriyaIITR** [comment]
Okay. But the output of the model and the loss is the same for a given random seed, no matter how many times you run it. but if I just change my target labels I don't understand how does it matter since all classes are assigned the same weight.

**Comment by ThilinaRajapakse** [comment]
Is the final loss and outputs different when you change only the label encoding?

**Comment by KrishnaPriyaIITR** [comment]
> Is the final loss and outputs different when you change only the label encoding?

Yes sir. That is the problem.

**Comment by ThilinaRajapakse** [comment]
What about the number of correct predictions vs number of incorrect predictions? Does that change too?
